x-pack/plugins/fleet/common/authz.ts

```diff
@@ -58,51 +63,86 @@ export interface FleetAuthz {
   };
 }
 
+interface ReadAllParams {
+  all: boolean;
+  read: boolean;
+}
+
 interface CalculateParams {
   fleet: {
     all: boolean;
     setup: boolean;
+    read?: boolean;
+    agents?: ReadAllParams;
+    agentPolicies?: ReadAllParams;
+    settings?: ReadAllParams;
   };
 
-  integrations: {
-    all: boolean;
-    read: boolean;
-  };
+  integrations: ReadAllParams;
 
-  isSuperuser: boolean;
+  subfeatureEnabled: boolean;
 }
 
 type PrivilegeMap = Record<string, { executePackageAction: boolean }>;
 
 export const calculateAuthz = ({
   fleet,
   integrations,
-  isSuperuser,
-}: CalculateParams): FleetAuthz => ({
-  fleet: {
-    all: fleet.all && (integrations.all || integrations.read),
-
-    // These are currently used by Fleet Server setup
-    setup: fleet.all || fleet.setup,
-    readEnrollmentTokens: fleet.all || fleet.setup,
-    readAgentPolicies: fleet.all || fleet.setup,
-  },
+  subfeatureEnabled,
+}: CalculateParams): FleetAuthz => {
+  const fleetAuthz: FleetAuthz['fleet'] = subfeatureEnabled
```

Comment:
it could be nice to add a comment / extract the objects to a variable to remind us to remove the fallback when the feature flag is removed

Reply:
👍 I will add a comment